### ISSUE TITLE
Incorrect characters in insta snapshots

### ISSUE THREAD

**Issue by tamasfe** [rhaiscript/lsp#102]
Weird characters appear seemingly randomly in formatter output snapshots that I haven't been able to reproduce outside tests.

![image](https://user-images.githubusercontent.com/25967296/188328903-409c2645-8eb7-49b2-b7b8-2562f9da6192.png)

**Comment by tamasfe** [comment]
I suspected that it was an use-after-free error due to the way handle strings in the formatter: https://github.com/rhaiscript/lsp/blob/c954855412d59702ea939c677f501ae28d9a3d29/crates/rhai-fmt/src/util.rs#L122

But miri does not complain about it.

**Comment by schungx** [comment]
> I suspected that it was an use-after-free error due to the way handle strings in the formatter:

I wonder how this can happen with Rust...  Is it unsafe code?

**Comment by tamasfe** [comment]
> > I suspected that it was an use-after-free error due to the way handle strings in the formatter:
> 
> I wonder how this can happen with Rust... Is it unsafe code?

Yes, what I linked is unsafe :). But it appears to be correct and isn't the cause.
